components/admin_console/admin_definition_plugins.jsx

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+export let cache = null;
+const adminDefinitionsPlugins = {};
+
+export function registerAdminConsolePlugin(pluginId, func) {
+    adminDefinitionsPlugins[pluginId] = func;
+    cache = null;
+}
+
+export function unregisterAdminConsolePlugin(pluginId) {
+    if (adminDefinitionsPlugins[pluginId]) {
+        delete adminDefinitionsPlugins[pluginId];
+        cache = null;
```

Comment:
at this point don't we need to reload/regenerate the new plugin definitions? Which would happen automatically if all of this lived inside the redux store :).

Reply:
or i guess maybe when the plugin unregisters.

Reply:
We can port everything here to be used through redux, I wanted to keep it as a simple data structure from the beginning, but I'm not sure if it is getting complex enough to move it to redux. I'll give it a try and see the result.